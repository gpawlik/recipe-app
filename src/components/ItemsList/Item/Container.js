import styled from 'styled-components';

import { colors } from 'variables';

const Container = styled.li`
  background-color: ${colors.WHITE};
  padding: 15px 25px;
  margin: 5px;
  display: flex;
  align-items: center;
`;

export default Container;
