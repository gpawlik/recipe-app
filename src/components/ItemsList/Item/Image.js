import styled from 'styled-components';

import { colors } from 'variables';

const Image = styled.div`
  border-radius: 50%;
  width: 75px;
  height: 75px;
  display: table;
  margin-right: 15px;
  background-image: url(${({ thumbnail }) => thumbnail});
  background-color: ${colors.DOVE_GRAY};
  background-size: cover;
`;

export default Image;
