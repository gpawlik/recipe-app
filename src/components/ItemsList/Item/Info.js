import styled from 'styled-components';

import { colors } from 'variables';

const Info = styled.p`
  font-size: 14px;
  color: ${colors.CONCRETE_GRAY};
  margin: 0;
`;

export default Info;
