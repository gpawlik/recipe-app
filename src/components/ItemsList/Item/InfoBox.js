import styled from 'styled-components';

const InfoBox = styled.div`
  margin: 5px;
`;

export default InfoBox;
