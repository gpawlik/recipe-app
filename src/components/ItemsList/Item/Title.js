import styled from 'styled-components';

import { colors } from 'variables';

const Title = styled.a`
  font-size: 20px;
  font-weight: 600;
  color: ${colors.MIDNIGHT_BLUE};
  text-decoration: none;
  margin-bottom: 5px;

  &:hover {
    color: ${colors.GREEN_SEA}
  }
`;

export default Title;
