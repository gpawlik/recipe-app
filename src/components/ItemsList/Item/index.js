import React from 'react';
import PropTypes from 'prop-types';

import Container from './Container';
import Image from './Image';
import InfoBox from './InfoBox';
import Title from './Title';
import Info from './Info';

import { decode } from 'utils/htmlEntities';

const Item = ({ title, thumbnail, href, ingredients }) => {
  return (
    <Container>
      <Image thumbnail={thumbnail} />
      <InfoBox>
        <Title href={href} target="_blank">{decode(title)}</Title>
        <Info>{decode(ingredients)}</Info>
      </InfoBox>
    </Container>
  );
};

Item.propTypes = {
  title: PropTypes.string,
  thumbnail: PropTypes.string,
  href: PropTypes.string,
  ingredients: PropTypes.string
};

export default Item;
