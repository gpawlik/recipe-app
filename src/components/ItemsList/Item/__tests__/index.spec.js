import React from 'react';
import { create } from 'react-test-renderer';

import Item from '../';

jest.mock('utils/htmlEntities', () => {
  return {
    decode: item => item
  };
});

const initialProps = {};

const mockComponent = props => {
  const finalProps = {
    ...initialProps,
    ...props
  };

  return (
    <Item {...finalProps} />
  );
};

describe('ItemsList/Item', () => {
  it('should render with the default props', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      title: 'foo',
      thumbnail: 'bar',
      href: 'baz',
      ingredients: 'faz'
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
