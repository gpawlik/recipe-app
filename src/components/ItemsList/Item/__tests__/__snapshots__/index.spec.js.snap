// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList/Item should render with custom props 1`] = `
<li
  className="sc-bdVaJa gRLtrj"
>
  <div
    className="sc-bwzfXH fGIKzA"
  />
  <div
    className="sc-htpNat dFmzvL"
  >
    <a
      className="sc-bxivhb EGuae"
      href="baz"
      target="_blank"
    >
      foo
    </a>
    <p
      className="sc-ifAKCX dqqmUa"
    >
      faz
    </p>
  </div>
</li>
`;

exports[`ItemsList/Item should render with the default props 1`] = `
<li
  className="sc-bdVaJa gRLtrj"
>
  <div
    className="sc-bwzfXH ccYzUD"
  />
  <div
    className="sc-htpNat dFmzvL"
  >
    <a
      className="sc-bxivhb EGuae"
      href={undefined}
      target="_blank"
    />
    <p
      className="sc-ifAKCX dqqmUa"
    />
  </div>
</li>
`;
